### PR TITLE
Capture user name with Azure SSO

### PR DESCRIPTION
**Pull request by SylvainJuge**
## What does this PR do?

When using Azure Tomcat service, the user name is not properly captured by Java agent.
This PR adds a fallback heuristic to capture the provided user name based on [official MS code samples](https://docs.microsoft.com/en-us/azure/app-service/configure-language-java?pivots=platform-linux#tomcat-1) and [documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens).

## Checklist

- [x] validate implementation manually with Azure/Tomcat deployment
- [x] This is an enhancement of existing features, or a new feature in existing plugins
  - [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [x] I have added tests that prove my fix is effective or that my feature works
  - [x] ~~Added an API method or config option? Document in which version this will be introduced~~ n/a
  - [x] ~~I have made corresponding changes to the documentation~~ n/a no doc change required.

### CODE REVIEW

File: apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java [hunk 1]
```diff
@@ -137,6 +139,44 @@ private void startCaptureBody(Transaction transaction, String method, @Nullable
...
+    @Nullable
+    private static String getFirstClaim(Map<?, ?> map, String key) {
+        // entries are stored in nested collection, even when there is a single entry
+        // https://docs.microsoft.com/en-us/azure/app-service/configure-language-java?pivots=platform-linux#tomcat-1
+        Object entry = map.get(key);
+        if (entry instanceof List) {
+            Object first = ((List<?>) entry).get(0);
+            if (first instanceof String) {
+                return (String) first;
+            }
```
Comment (eyalkoren) [inline apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java:175]: [minor] if we find a value for the looked-up key, don't we want to always use it?
```suggestion
            if (first != null) {
                first.toString();
            }
```

File: apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java [hunk 2]
```diff
@@ -137,6 +139,44 @@ private void startCaptureBody(Transaction transaction, String method, @Nullable
...
+
+        return userName;
+    }
+
+    @Nullable
+    private static String getFirstClaim(Map<?, ?> map, String key) {
+        // entries are stored in nested collection, even when there is a single entry
+        // https://docs.microsoft.com/en-us/azure/app-service/configure-language-java?pivots=platform-linux#tomcat-1
+        Object entry = map.get(key);
+        if (entry instanceof List) {
```
Comment (eyalkoren) [inline apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java:171]: [questions] 
If it is not a list we don't want it? 
Is it never possible to be something else?
Can the obtained list be empty, making `get(0)` throw `IndexOutOfBoundsException`?

**Review by eyalkoren — APPROVED** [review]
Thanks!
Minor proposal and questions - please decide if they require addition/change/more-checks based on what you know.

File: apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java (hunk 2 above)
Comment (SylvainJuge) [reply apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletTransactionHelper.java:171]: good point on checking the list size, I'll add that.
The only information I have on those collections comes from experimentation, the code samples refer to those as generic `Collection` but they are actually `ArrayList` so we can avoid creating an iterator, I have not seen any entry which is not a list, so I think we can revisit this later if that proves to be an issue.

**Pull request merged**